server/users/service.go

```diff
@@ -254,10 +266,18 @@ func (rs *RecognitionService) runPipeline(destinationID string, identifiers Even
 			continue
 		}
 
-		err = rs.metaStorage.DeleteAnonymousEvent(destinationID, identifiers.AnonymousID, storedEventID)
-		if err != nil {
-			logging.SystemErrorf("[%s] Error deleting stored recognized event [%s]: %v", destinationID, storedEventID, err)
-			continue
+		// If any property is empty event will be anonymous until filling out that last property.
+		// Removing event from storage is suitable only when all properties are filled out.
+		if configuration.IdentificationJSONPathes.IsFullFilled(event) {
```

Comment:
This condition could be omitted because `runPipeline()` call happens only if All `identification_nodes` are present.
(IsFullFilled() func should be deleted from JSONPathes struct).

You can just keep the old piece of code here.

Reply:
Condition was removed